packages/toolpad-app/src/ThemeContext.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import Head from 'next/head';
+import { ThemeProvider as MuiThemeProvider, createTheme } from '@mui/material/styles';
+import { deepmerge } from '@mui/utils';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { getDesignTokens, getThemedComponents, getMetaThemeColor } from './theme';
+import useLocalStorageState from './utils/useLocalStorageState';
+
+interface ThemeProviderProps {
+  children?: React.ReactNode;
+}
+
+type PaletteMode = 'light' | 'dark';
+export type ThemeMode = PaletteMode | 'system';
+
+function usePreferredMode(): PaletteMode {
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+  return prefersDarkMode ? 'dark' : 'light';
+}
+
+export function useThemeMode() {
+  return useLocalStorageState<ThemeMode>('toolpad-palette-mode', 'system');
```

Comment:
To make it more self documenting for consumers of this hook, maybe we should have it return a

```suggestion
  const [theme, setTheme] = useLocalStorageState<ThemeMode>('toolpad-palette-mode', 'system');
  return { theme, setTheme };
```
?